### PR TITLE
refactor(RootService): remove app references.

### PR DESCRIPTION
**Pull request by bchrisarriola**
There are still a couple remaining but those methods still have to be migrated

### CODE REVIEW

File: Blockchain/URL/BitcoinURLPayload.swift [hunk 1]
```diff
@@ -30,7 +35,9 @@ extension BitcoinURLPayload {
 
         let queryArgs = url.queryArgs
 
-        self.address = url.host ?? queryArgs["address"]
-        self.amount = queryArgs["amount"]
+        let address = url.host ?? queryArgs["address"]
```
Comment (woudini) [inline Blockchain/URL/BitcoinURLPayload.swift]: `address` is being set to nil with a `URL` of `bitcoin:1Amu4uPJnYbUXX2HhDFMNq7tSneDwWYDyv` (web format). Should probably add support for the `bitcoin:` format and add it to `BitcoinURLPayload` tests.

Comment (bchrisarriola) [reply Blockchain/URL/BitcoinURLPayload.swift]: Good catch. Can ether or bitcoincash also be formatted in that way?

File: Blockchain/URL/BitcoinURLPayload.swift [hunk 2]
```diff
@@ -9,17 +9,22 @@
...
+        self.address = address
+        self.amount = amount
+    }
 }
 
 extension BitcoinURLPayload {
-    init?(url: URL) {
+    @objc convenience init?(url: URL) {
         guard let scheme = url.scheme else {
             return nil
```
Comment (woudini) [inline Blockchain/URL/BitcoinURLPayload.swift]: This `guard` statement and the one below it are a bit too strict. Ideally the scanner should be "smart" enough to allow both `bitcoin:address` and just `address`, and the same with BCH. Sorry I should have mentioned earlier.

Comment (bchrisarriola) [reply Blockchain/URL/BitcoinURLPayload.swift]: I don't think it's unreasonable to expect that a URL should have a scheme using this initializer. In the case of just `address`, you can always use the main initializer `BitcoinURLPayload(address: address, amount: amount)`.

Comment (woudini) [reply Blockchain/URL/BitcoinURLPayload.swift]: It's fine where it is but the scanner is too strict. Users will complain when they can't scan a BTC/BCH address with no `bitcoin:` prefix.

**Pull request merged**